### PR TITLE
Add to Inbound from master list : UI

### LINKED ISSUE #365: Inbound: Add from master list
Implement #117 ( and  #274 and have a look at the list in #302 ) for Inbounds

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #365 

Adds the button, hooks up the api, updates outbound to use the new mutation.
Have made the unallocated lines blue and hidden columns, as per the outbound implementation.

## Tests
- [ ] New 'Add from master list' button
- [ ] Errors show as a toast
- [ ] Adding works
- [ ] Lines show as blue and columns are mostly blank, until a quantity is entered

### CODE REVIEW

File: client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts
Comment (clemens-msupply) [inline client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts:19]: numberOfPacks is non-optional so IMHO more readable to do `row.numberOfPacks === 0`

**Review by clemens-msupply — COMMENTED** [review]
Looks good! (haven't tested yet though)

File: client/packages/invoices/src/InboundShipment/DetailView/AddFromMasterListButton.tsx
Comment (mark-prins) [reply client/packages/invoices/src/InboundShipment/DetailView/AddFromMasterListButton.tsx:29]: We have all of the modals mounting on page load currently, and I'd prefer to continue with that approach ( unless necessary to change.. like the patient modal! ). This is how material UI have implemented modals, and can help with  faster loading

File: client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts
Comment (mark-prins) [reply client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts:19]: yeah, I agree. thanks - had taken the nullable from the outbound 'unallocated lines' which is a different scenario.

File: client/packages/invoices/src/InboundShipment/DetailView/AddFromMasterListButton.tsx
Comment (clemens-msupply) [reply client/packages/invoices/src/InboundShipment/DetailView/AddFromMasterListButton.tsx:29]: cool thanks

**Comment by andreievg** [comment]
Haven't quite looked at the PR yet, but just something from reviewing back end PR:

Add from master list for both inbound and outbound should only be allowed when invoice status is `new` (for outbound) and `new` or potentially `shipped` (for inbound). And perhaps we need a warning that unallocated or 0 lines will be trimmed when moving from those two statuses

**Comment by mark-prins** [comment]
Thanks @andreievg - good points. yes, was planning to put those in.. was just waiting until the b/e PR was merged before reworking this one. I'll have to get those changes merged into this branch

**Comment by mark-prins** [comment]
updated with the b/e changes, and have made the changes noted above. thanks for the reveiws!

File: client/packages/invoices/src/InboundShipment/DetailView/ContentArea/ContentArea.tsx
Comment (andreievg) [inline client/packages/invoices/src/InboundShipment/DetailView/ContentArea/ContentArea.tsx:35]: does it need to be exported ?

**Comment by andreievg** [comment]
just testing now, do we want to make default sort by item name ?

**Comment by andreievg** [comment]
Small one for later, if you edit 'empty' line without changing quantity it is removed, I think it's desired behaviour, but I think it should be auto trimmed or manually deleted, should allow 0 (and only really trim on verified). But that's too much to change right now)

File: client/packages/invoices/src/InboundShipment/DetailView/ContentArea/ContentArea.tsx
Comment (mark-prins) [reply client/packages/invoices/src/InboundShipment/DetailView/ContentArea/ContentArea.tsx:35]: ugh. how did that slip in there? thanks

**Comment by andreievg** [comment]
was about to approve, but can't seem to go to allocated status when one line is allocated but other not in OS, i thought we trim them, i dont' think it's something youv'e done in this pr

**Comment by andreievg** [comment]
Ok did some test, I thi

> was about to approve, but can't seem to go to allocated status when one line is allocated but other not in OS

False alarm, i had above 0 quantity of allocated line (althought something to flag, it was hard to know why i was getting this error)

**Review by andreievg — APPROVED** [review]
Approving, but I think we should limit outbound shipment add from master list for new status only, IS is working very good, thank you

**Comment by mark-prins** [comment]
thanks for the review! agree about the outbounds, will make another issue for that one. And for only adding items with stock lines to the IS. The other changes are done

**Pull request merged**
